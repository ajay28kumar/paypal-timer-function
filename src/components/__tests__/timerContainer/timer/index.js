import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Timer from "../../../timerContainer/timer";

const mockDeleteFunction = jest.fn();

const props = {
  startCounter: 56,
  serialNumber: 1,
  deleteTimer: mockDeleteFunction,
};

describe("test Timer component", () => {
  let wrapper = shallow(<Timer {...props} />);
  it("snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
