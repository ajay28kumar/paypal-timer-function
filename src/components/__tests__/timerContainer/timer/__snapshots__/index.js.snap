// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Timer component snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
  raised={true}
>
  <div
    className="makeStyles-deleteContainer-5"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-serialCount-2"
      color="primary"
    >
      1
    </WithStyles(ForwardRef(Typography))>
    <Memo(ForwardRef(DeleteOutlinedIcon))
      className="cursor-pointer"
      color="error"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="makeStyles-container-3"
  >
    56
  </div>
  <div
    className="makeStyles-buttonContainer-4"
  >
    <WithStyles(ForwardRef(Button))
      color="secondary"
      disabled={false}
      id="stop-button"
      onClick={[Function]}
      variant="contained"
    >
      Stop
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      id="restart-button"
      onClick={[Function]}
      variant="outlined"
    >
      Restart
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      id="start-button"
      onClick={[Function]}
      variant="contained"
    >
      Pause
    </WithStyles(ForwardRef(Button))>
  </div>
</WithStyles(ForwardRef(Card))>
`;
